addons/website_sale/controllers/main.py

```diff
@@ -160,13 +160,24 @@ def _get_search_order(self, post):
         # id is added to be sure that order is a unique sort key
         return 'is_published desc,%s , id desc' % post.get('order', 'website_sequence desc')
 
-    def _get_search_domain(self, search, category, attrib_values):
+    def _get_search_domain(self, search, category, attrib_values, search_in_description):
         domain = request.website.sale_product_domain()
         if search:
             for srch in search.split(" "):
-                domain += [
-                    '|', '|', '|', ('name', 'ilike', srch), ('description', 'ilike', srch),
-                    ('description_sale', 'ilike', srch), ('product_variant_ids.default_code', 'ilike', srch)]
+                if search_in_description:
+                    domain += [
+                        '|', '|', '|',
+                        ('name', 'ilike', srch),
+                        ('product_variant_ids.default_code', 'ilike', srch),
+                        ('description', 'ilike', srch),
+                        ('description_sale', 'ilike', srch)
+                    ]
+                else:
+                    domain += [
+                        '|',
+                        ('name', 'ilike', srch),
+                        ('product_variant_ids.default_code', 'ilike', srch)
+                    ]
```

Comment:
Duplicated code. Add the non description part in the doman, then if search_in_description is True, add the description part. Also make the search_in_description parameter True by default (as it was before).

So
```python
..., search_in_description=True):
```
```python
domain += [...]
if search_in_description:
    domain += [...]
```

Reply:
That's what I did at first for domain but It does not work. What am I doing wrong ?
```
                domain += [
                    '|',
                    ('name', 'ilike', srch),
                    ('product_variant_ids.default_code', 'ilike', srch)
                ]
                if search_in_description:
                    domain += [
                        '|',
                        ('description', 'ilike', srch),
                        ('description_sale', 'ilike', srch)
                    ]
```
This said, I don't understand why search_in_description should be True by default ?

Reply:
In general, so that people who used the function in their code do not need to adapt their code for nothing (as you add a required parameter, their python code would crash). But ok, as it a private function which is only used once by shop, you can leave it without setting True by default.

And your domain does not work because it is wrong. See you have 3 `|` in your previous code and now only 2. Indeed, you final domain is : `['|', A, B, '|', C, D]` which equal to `['&', '|', A, B, '|', C, D]`, so you are saying "give me the product which contain X in the name AND in the description, so probably none. What you want is `['|', '|', A, B, '|', C, D]`. In fact to achieve this, instead of relying on the + operator, check how to build domains with `expression.OR` and `expression.AND`.